app/components/course-page/share-progress-modal.ts

```diff
@@ -0,0 +1,71 @@
+import type AnalyticsEventTrackerService from 'codecrafters-frontend/services/analytics-event-tracker';
+import Component from '@glimmer/component';
+import config from 'codecrafters-frontend/config/environment';
+import fade from 'ember-animated/transitions/fade';
+import RepositoryModel from 'codecrafters-frontend/models/repository';
+import { action } from '@ember/object';
+import { inject as service } from '@ember/service';
+import { task, timeout } from 'ember-concurrency';
+import { tracked } from '@glimmer/tracking';
+
+interface Signature {
+  Element: HTMLDivElement;
+
+  Args: {
+    repository: RepositoryModel;
+    onClose: () => void;
+  };
+
+  Blocks: {
+    default: [];
+  };
+}
+
+export default class ShareProgressModalComponent extends Component<Signature> {
+  fade = fade;
+
+  socialPlatforms = ['twitter', 'slack', 'discord', 'linkedin'];
+
+  @service declare analyticsEventTracker: AnalyticsEventTrackerService;
+
+  @tracked selectedSocialPlatform = 'twitter';
+  @tracked wasCopiedRecently = false;
+
+  get copyableText() {
```

Comment:
@libmartinito we should allow editing the copyable text (that's why it is a textarea). The idea is that users will edit the text and then copy

Reply:
If they click one of the icons, we can reset the text to the default for that platform